ltiauthenticator/__init__.py

```diff
@@ -184,5 +186,19 @@ class LTIAuthenticateHandler(BaseHandler):
 
     @gen.coroutine
     def post(self):
+        # References:
+        # - Class dependencies:
+        #   - jupyterhub.handlers.BaseHandler: https://github.com/jupyterhub/jupyterhub/blob/abb93ad799865a4b27f677e126ab917241e1af72/jupyterhub/handlers/base.py#L69
+        #   - tornado.web.RequestHandler: https://www.tornadoweb.org/en/stable/web.html#tornado.web.RequestHandler
+        # - Function dependencies:
+        #   - login_user: https://github.com/jupyterhub/jupyterhub/blob/abb93ad799865a4b27f677e126ab917241e1af72/jupyterhub/handlers/base.py#L696-L715
+        #   - get_next_url: https://github.com/jupyterhub/jupyterhub/blob/abb93ad799865a4b27f677e126ab917241e1af72/jupyterhub/handlers/base.py#L587
+        #   - get_body_argument: https://www.tornadoweb.org/en/stable/web.html#tornado.web.RequestHandler.get_body_argument
         user = yield self.login_user()
-        self.redirect(self.get_body_argument('custom_next', self.get_next_url()))
+        next_url = self.get_next_url(user=user)
+        body_argument = self.get_body_argument(
+            name='custom_next',
+            default=next_url,
+        )
```

Comment:
## Considerations
- [x] The user variable was never considered, should it be? The get_next_url has logic to consider a passed user object as a fallback to other logic. While I'm not confident this will or won't be relevant. I think it could be a bugfix of #11 to pass the user to get_next_url.
- [x] I named parameters passed to get_body_argument to make it easier to read what it is about. I also verified that this was the naming some versions back as well to avoid potentially making this compatible by being explicit here.

Reply:
I think a bugfix for #11 would make this a major incompatible release, since users might rely on that now. I think if we do fix it, we should put it behind a flag.

Reply:
If we know this isn't going to change behavior around #11, I am happy to merge this now

Reply:
I dont get the outcome of passing user to get_next_route fully, other than it allows for a fallback for the returned value in the end of that function.

I think it may make sense but i lack knowledge about this.

i can revert this change of passing user to the get_next_route and open a discussion about it or another pr for that specifically.

i dont think i can get a proper understanding about this myself, i lack an experimental setup

Reply:
I've added a commit where I stop passing user to `get_next_url` again which should make this PR safe to merge again.

Reply:
I just looked at https://github.com/jupyterhub/jupyterhub/blob/master/jupyterhub/handlers/base.py#L587, and in our case passing user= is a no-op. I'll merge this then :)